app/api/routers/breeze_buddy.py

```
@@ -702,3 +709,112 @@ async def get_call_details(
+
+            # Fetch the audio file
+            logger.info(f"Fetching recording from URL: {recording_url}")
+            async with session.get(recording_url, headers=headers) as response:
+                if response.status != 200:
+                    logger.error(
+                        f"Failed to fetch recording. Status: {response.status}, URL: {recording_url}"
+                    )
+                    raise HTTPException(
+                        status_code=response.status,
+                        detail=f"Failed to fetch recording from provider. Status: {response.status}"
+                    )
```

Ruff (0.14.8)

Abstract `raise` to an inner function

(TRY301)